contracts/upgradeable_contracts/amb_erc677_to_erc677/HomeStakeTokenMediator.sol

```diff
@@ -0,0 +1,160 @@
+        return
+            super.initialize(
+                _bridgeContract,
+                _mediatorContract,
+                _erc677token,
+                _dailyLimitMaxPerTxMinPerTxArray,
+                _executionDailyLimitExecutionMaxPerTxArray,
+                _requestGasLimit,
+                _decimalShift,
+                _owner
+            );
+    }
+
+    /**
+     * Initializes rewardable home mediator
+     * @param _bridgeContract HomeAMB bridge contract
+     * @param _mediatorContract address of the mediator contract in the Foreign chain
+     * @param _erc677token address of STAKE token in the Home chain
+     * @param _dailyLimitMaxPerTxMinPerTxArray Home limits for outgoing transfers
+     * @param _executionDailyLimitExecutionMaxPerTxArray Home execution limits for incoming transfers
+     * @param _requestGasLimit gas limit used for AMB operations
+     * @param _decimalShift decimal shift for bridged TAKE token
+     * @param _owner address of new bridge owner
+     * @param _blockReward address of block reward contract used for fee distribution
+     * @param _fee initial home fee
+     */
+    function rewardableInitialize(
+        address _bridgeContract,
+        address _mediatorContract,
+        address _erc677token,
+        uint256[] _dailyLimitMaxPerTxMinPerTxArray, // [ 0 = _dailyLimit, 1 = _maxPerTx, 2 = _minPerTx ]
+        uint256[] _executionDailyLimitExecutionMaxPerTxArray, // [ 0 = _executionDailyLimit, 1 = _executionMaxPerTx ]
+        uint256 _requestGasLimit,
+        uint256 _decimalShift,
+        address _owner,
+        address _blockReward,
+        uint256 _fee
+    ) external returns (bool) {
+        _setFee(_fee);
+        _setBlockRewardContract(_blockReward);
+        return
+            initialize(
+                _bridgeContract,
+                _mediatorContract,
+                _erc677token,
+                _dailyLimitMaxPerTxMinPerTxArray,
+                _executionDailyLimitExecutionMaxPerTxArray,
+                _requestGasLimit,
+                _decimalShift,
+                _owner
+            );
+    }
+
+    /**
+     * @dev Allows to transfer token ownership to different proxy contract.
+     * Can be called only once, when mediator is the current owner of a token.
+     * All subsequent calls to erc677 token will be done through new proxy contract.
+     * @param _owner token proxy contract address
+     */
+    function transferTokenOwnership(address _owner) external onlyOwner {
+        Ownable(erc677token()).transferOwnership(_owner);
+    }
+
+    /**
+     * @dev Updates address of contract used for handling mint operations,
+     * all subsequent mint operations will be called through this contract
+     * @param _mintHandler address of new contract
+     */
+    function setMintHandler(address _mintHandler) external onlyOwner {
+        require(AddressUtils.isContract(_mintHandler));
+        addressStorage[MINT_HANDLER] = _mintHandler;
+    }
```

Comment:
Please clarify, why do we need to have a separate mint handler? In which cases will it differ from `erc677token()`?

Reply:
In the deployed STAKE token contract for the home chain, only owner is allowed to mint tokens.

For one bridge, we simply call `transferOwnership(mediator)` on token. Mediator uses `erc677token()` for minting.

In case of several bridges for the same token, all of them should be able to mint tokens. For that, we introduce a special contract that will be the owner of the token. This contract is also set to be a `mintHandler` in all existing bridges. This contract will be a simple proxy between bridges and a token for mint operations. A draft example of such contract can be found here: https://github.com/poanetwork/tokenbridge-contracts/blob/develop/contracts/mocks/MintHandlerMock.sol
It is used in tests for mint handler functionality.

Reply:
Got it, thanks.